client/image_push.go

```diff
@@ -25,15 +25,14 @@ func (cli *Client) ImagePush(ctx context.Context, image string, options types.Im
 		return nil, errors.New("cannot push a digest reference")
 	}
 
-	tag := ""
 	name := reference.FamiliarName(ref)
-
-	if nameTaggedRef, isNamedTagged := ref.(reference.NamedTagged); isNamedTagged {
-		tag = nameTaggedRef.Tag()
-	}
-
 	query := url.Values{}
-	query.Set("tag", tag)
+	if !options.All {
```

Comment:
I was considering if we should produce an error if `options.All` is set and `image` contains a tag; currently this situation is handled in the CLI;

```go
	case opts.all && !reference.IsNameOnly(ref):
 		return errors.New("tag can't be used with --all-tags/-a")
```

Which is the same as how it's handled for `docker pull`, but perhaps we want that case to be handled by the client instead.

Also happy to do that in a follow-up and do it for both `Push` and `Pull` if we want them to act consistent.